src/brogue/MainMenu.c

```diff
@@ -237,136 +247,372 @@ void initializeMenuFlames(boolean includeTitle,
+        initializeMainMenuButton(&(buttons[1]), "  New %sS%seeded Game  ", 's', 'S', NG_NEW_GAME_WITH_SEED);
+        initializeMainMenuButton(&(buttons[2]), "     %sL%soad Game     ", 'l', 'L', NG_OPEN_GAME);
+
+    } else if (rogue.nextGame == NG_FLYOUT_VIEW) {
+
+        buttonCount = 2;
+        initializeMainMenuButton(&(buttons[0]), "   View %sR%secording  ", 'r', 'R', NG_VIEW_RECORDING);
+        initializeMainMenuButton(&(buttons[1]), "    %sH%sigh Scores    ", 'h', 'H', NG_HIGH_SCORES);
+
+    } else if (rogue.nextGame == NG_FLYOUT_OPTIONS) {
+
+        buttonCount = 2;
+        initializeMainMenuButton(&(buttons[0]), "    Game V%sa%sriant   ", 'a', 'A', NG_GAME_VARIANT);
+        initializeMainMenuButton(&(buttons[1]), "     Game %sM%sode     ", 'm', 'M', NG_GAME_MODE);
+
+    } else {
+        return;
+    }
+
+    stackButtons(buttons, buttonCount, position, 2, false);
+    initializeMenu(menu, buttons, buttonCount, shadowBuf);
+}
+
+/// @brief Displays a dialog window for the user to chose a game variant
+static void chooseGameVariant() {
+    short gameVariantChoice;
+    char textBuf[COLS * ROWS] = "", tmpBuf[COLS * ROWS] = "", goldColorEscape[5] = "", whiteColorEscape[5] = "";
+
+    encodeMessageColor(goldColorEscape, 0, &yellow);
+    encodeMessageColor(whiteColorEscape, 0, &white);
+
+    sprintf(tmpBuf, "%sBrogue%s\n", goldColorEscape, whiteColorEscape);
+    strcat(textBuf, tmpBuf);
+    strcat(textBuf, "Classic Brogue. The endlessly captivating masterpiece of dungeon adventuring.\n\n");
+
+    sprintf(tmpBuf, "%sRapid Brogue%s\n", goldColorEscape, whiteColorEscape);
+    strcat(textBuf, tmpBuf);
+    strcat(textBuf, "Die faster and more often in this quarter-length version of the classic game!\n\n");
+
+    brogueButton buttons[2];
+    cellDisplayBuffer rbuf[COLS][ROWS];
+    copyDisplayBuffer(rbuf, displayBuffer);
+    initializeMainMenuButton(&(buttons[0]), "  %sR%sapid Brogue     ", 'r', 'R', NG_NOTHING);
+    initializeMainMenuButton(&(buttons[1]), "     %sB%srogue        ", 'b', 'B', NG_NOTHING);
+    gameVariantChoice = printTextBox(textBuf, 20, 7, 45, &white, &black, rbuf, buttons, 2);
+    overlayDisplayBuffer(rbuf, NULL);
+
+    if (gameVariantChoice == 1) {
+        gameVariant = VARIANT_BROGUE;
+    } else if (gameVariantChoice == 0) {
+        gameVariant = VARIANT_RAPID_BROGUE;
+    } else {
+        rogue.nextGame = NG_NOTHING;
+    }
+}
+
+/// @brief Displays a dialog window for the user to chose a game mode. The game mode is displayed in the bottom left
+/// on the title screen (except normal mode).
+static void chooseGameMode() {
+    short gameMode;
+    char textBuf[COLS * ROWS] = "", tmpBuf[COLS * ROWS] = "", goldColorEscape[5] = "", whiteColorEscape[5] = "";
+
+    encodeMessageColor(goldColorEscape, 0, &yellow);
+    encodeMessageColor(whiteColorEscape, 0, &white);
+
+    sprintf(tmpBuf, "%sNormal Mode%s\n", goldColorEscape, whiteColorEscape);
```

Comment:
Likewise, prefer `snprintf`

Reply:
Done. Also I switched to the append function for string building.